deployment.md

```diff
@@ -0,0 +1,162 @@
+However, it is possible to operate with only 1 liquidity pool, Token <> ETH. In this case, one oracle will provide the Token <> ETH price, and the other should state the exchange rate of USD <> USD, and therefore can be a fixed oracle.
+
+#### Liqiudity pool
+
+Token <> ETH liquidity pool is required for the operation of the ERC20 paymaster. It is used by the TwapOracle to provide the price feed for the ERC20 paymaster.
+Liqiudity pool deployment is outside the scope of this document.
+
+### Factories
+
+Deployment of the paymaster and its infrastructure happens through the corresponding factory contracts on each chain.
+
+Note that for security purposes, both factories are ownerful, and only the owner can deploy new instances of the paymaster and its infrastructure.
+Therefore, you need to specify the owner address when deploying the factory. You should store the owner address to `$FACTORY_OWNER_ADDRESS` and private key for further usage to `$FACTORY_OWNER_PRIVATE_KEY`.
+You can deploy the factory with any account, so specify its private key in `$DEPLOYER_PRIVATE_KEY`.
+
+To deploy the Oracle factory, run the following command:
+
+```shell
+forge create --optimizer-runs 100 -via-ir src/factory/ERC20OracleOnlyFactory.sol:ERC20OracleOnlyFactory --constructor-args $FACTORY_OWNER_ADDRESS --private-key $DEPLOYER_PRIVATE_KEY -r $PROVIDER_URL -c <chain_id>
+```
+
+The command will return the address of the deployed factory, that you should save to `$ORACLE_FACTORY_ADDRESS`.
+
+---
+
+To deploy the Paymaster factory, run the following command:
+
+```shell
+forge create --optimizer-runs 100 -via-ir src/factory/ERC20PaymasterOnlyFactory.sol:ERC20PaymasterOnlyFactory --constructor-args $FACTORY_OWNER_ADDRESS --private-key $DEPLOYER_PRIVATE_KEY -r $PROVIDER_URL -c <chain_id>
+```
+
+The command will return the address of the deployed factory, that you should save to `$PAYMASTER_FACTORY_ADDRESS`.
+
+## Deployment
+
+Below commands require some parameters and environment variables:
+
+- <salt> - the random bytes32 value that is used to derive the address of the deployed contract. It can be unique for each deployment.
+- <chain_id> - the chain id of the network where the contract will be deployed.
+- `$NATIVE_POOL_ADDRESS` - the address of the Token <> ETH liquidity pool.
+- `$POOL_NATIVE_TOKEN_ADDRESS` - the address of the native token FROM THE LIQUIDITY POOL.
+- `$TOKEN_ADDRESS` - the address of the token that the paymaster will operate with.
+- `$PAYMASTER_OWNER_ADDRESS` - the address of the paymaster owner, that will be able to change the price markup.
+- <price_markup> - the price markup, where 1000000 means 100%. It is used to calculate the price of the token in the paymaster. For example, 100% markup and that the user will pay at the actual price, and 120% means the user pays 20% more. Possible values are from 0 to 4294967295.
+- <max_price_markup> - the maximum price markup that the paymaster owner can set.
```

Comment:
I suggest adding the same example in numbers as for `price_markup` point above since it may be confusing for those who are not "into paymasters" that price markup is a percentage but *max* price markup is just a token amount in wei

Reply:
60879dbb5b1bb13b48bb5bc131b6e990d3862a88